From f9403232bb458972f10554b0e6a178cc563cf889 Mon Sep 17 00:00:00 2001
From: Tim S <stahta01@users.sourceforge.net>
Date: Fri, 16 Mar 2018 18:44:19 -0400
Subject: [PATCH] Test for MINGW64_VERSION_MAJOR

---
 include/wx/filefn.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/wx/filefn.h b/include/wx/filefn.h
index ca6f7ea142..f05eb6d53e 100644
--- a/include/wx/filefn.h
+++ b/include/wx/filefn.h
@@ -210,7 +210,7 @@ enum wxFileKind
     // to avoid using them as they're not present in earlier versions and
     // always using the native functions spelling is easier than testing for
     // the versions
-    #if defined(__BORLANDC__) || defined(__DMC__) || defined(__WATCOMC__) || defined(__MINGW64__)
+    #if defined(__BORLANDC__) || defined(__DMC__) || defined(__WATCOMC__) || defined(__MINGW64_VERSION_MAJOR)
         #define wxPOSIX_IDENT(func)    ::func
     #else // by default assume MSVC-compatible names
         #define wxPOSIX_IDENT(func)    _ ## func
@@ -245,7 +245,7 @@ enum wxFileKind
     #endif
 
     #ifdef wxHAS_HUGE_FILES
-        #ifndef __MINGW64__
+        #ifndef __MINGW64_VERSION_MAJOR
             #define   wxSeek       wxPOSIX_IDENT(lseeki64)
             #define   wxLseek      wxPOSIX_IDENT(lseeki64)
             #define   wxTell       wxPOSIX_IDENT(telli64)
-- 
2.16.2.windows.1

