From aac1987a9a37b71ffd7436dc103e7cf7bc4753d7 Mon Sep 17 00:00:00 2001
From: Tim S <tim.stahlhut@gmail.com>
Date: Sun, 10 Feb 2019 18:52:13 -0500
Subject: code fix: add DWARF2_UNWIND_INFO guard

---
 gcc/final.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gcc/final.c b/gcc/final.c
index c4812eac090..92fd4757aba 100644
--- a/gcc/final.c
+++ b/gcc/final.c
@@ -2207,10 +2207,12 @@ final_scan_insn (rtx insn, FILE *file, int optimize ATTRIBUTE_UNUSED,
 	    location_t loc;
 	    expanded_location expanded;
 
+#if defined (DWARF2_UNWIND_INFO)
 	    /* Make sure we flush any queued register saves in case this
 	       clobbers affected registers.  */
 	    if (dwarf2out_do_frame ())
 	      dwarf2out_frame_debug (insn, false);
+#endif
 
 	    /* There's no telling what that did to the condition codes.  */
 	    CC_STATUS_INIT;
-- 
