From 2564a2e525945f6eceb6f956d0f56ce9046d05f1 Mon Sep 17 00:00:00 2001
From: Tim Stahlhut <stahta01@gmail.com>
Date: Fri, 31 May 2019 00:34:00 -0400
Subject: Edit chkconf.h

---
 include/wx/gtk/chkconf.h | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/include/wx/gtk/chkconf.h b/include/wx/gtk/chkconf.h
index 84595b256c..4c59f41650 100644
--- a/include/wx/gtk/chkconf.h
+++ b/include/wx/gtk/chkconf.h
@@ -51,13 +51,6 @@
 
 #ifndef __UNIX__
 
-#   undef  wxUSE_WEBVIEW
-#   define wxUSE_WEBVIEW 0
-#   undef  wxUSE_WEBVIEW_WEBKIT
-#   define wxUSE_WEBVIEW_WEBKIT 0
-
-#   undef  wxUSE_MEDIACTRL
-#   define wxUSE_MEDIACTRL 0
 
     /*
         We could use GDK_WINDOWING_X11 for those but this would require
-- 
