From 8abb5e83a273c3bec5258d78e766c154b78dee31 Mon Sep 17 00:00:00 2001
From: Tim Stahlhut <stahta01@gmail.com>
Date: Sat, 10 Aug 2019 09:06:38 -0400
Subject: Add wxCHECK_MINGW64_VERSION

---
 include/wx/msw/gccpriv.h | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/include/wx/msw/gccpriv.h b/include/wx/msw/gccpriv.h
index d0b47e4183..1a6977bfe0 100644
--- a/include/wx/msw/gccpriv.h
+++ b/include/wx/msw/gccpriv.h
@@ -65,6 +65,14 @@
         #endif
     #endif
 
+    #ifdef __MINGW64_TOOLCHAIN__
+        #define wxCHECK_MINGW64_VERSION( major, minor ) \
+ ( ( ( __MINGW64_MAJOR_VERSION > (major) ) \
+      || ( __MINGW64_MAJOR_VERSION == (major) && __MINGW64_MINOR_VERSION >= (minor) ) ) )
+    #else
+        #define wxCHECK_MINGW64_VERSION( major, minor ) (0)
+    #endif
+
     #define wxCHECK_MINGW32_VERSION( major, minor ) \
  ( ( ( __MINGW32_MAJOR_VERSION > (major) ) \
       || ( __MINGW32_MAJOR_VERSION == (major) && __MINGW32_MINOR_VERSION >= (minor) ) ) )
-- 
