From: Tim Stahlhut <stahta01@gmail.com>
Date: Mon, 25 Nov 2019 11:56:12 -0500
Subject: Edit setup.h

---
 wx/setup.h           | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/wx/setup.h b/wx/setup.h
index 764e9a4..584e096 100644
--- a/wx/setup.h
+++ b/wx/setup.h
@@ -243,7 +243,7 @@
 
 #define wxUSE_INTL 1
 
-#define wxUSE_XLOCALE       0
+#define wxUSE_XLOCALE       1
 
 #define wxUSE_DATETIME 1
 
-- 
