From d40efcf4a7161b6b3b1da86c68a8b44c19947a07 Mon Sep 17 00:00:00 2001
From: Tim S <stahta01@users.sourceforge.net>
Date: Tue, 20 Mar 2018 00:43:59 -0400
Subject: [PATCH 17/22] wxWidgets3.0: Add conditional ToStdString() call

---
 src/Language.cpp                            | 4 ++++
 1 files changed, 4 insertions(+)

diff --git a/src/Language.cpp b/src/Language.cpp
index 5becf84..71bd314 100644
--- a/src/Language.cpp
+++ b/src/Language.cpp
@@ -138,7 +138,11 @@ LANGUAGE_ERROR compile_language(string lang_in, string lang_cache) {
 }
 
 void try_to_load_language( ) {
+#if wxCHECK_VERSION(3, 0, 0)
+	if (LANGERR_OK!=load_language(DIR_PLUS_FILE("lang",config->Init.language_file).ToStdString(),config->GetUserConfigPath("lang_cache").ToStdString())) {
+#else
 	if (LANGERR_OK!=load_language(DIR_PLUS_FILE("lang",config->Init.language_file).c_str(),config->GetUserConfigPath("lang_cache").c_str())) {
+#endif
 		if (g_splash) g_splash->Hide(); // en window, si el splash esta visible, la llamada a ShowModal revienta
 		mxMessageDialog(nullptr,"No se pudo cargar el diccionario del idioma seleccionado.\n"
 						"El sistema utilizará el predeterminado (español).\n\n"
-- 
2.16.2.windows.1

