From a422f6261976716789a6411af33f8a7b586b2ae3 Mon Sep 17 00:00:00 2001
From: Tim S <stahta01@users.sourceforge.net>
Date: Sun, 18 Mar 2018 10:31:26 -0400
Subject: [PATCH 10/21] * code: Add IsEmpty to fix wxWidgets 3.0

---
 src/ProjectManager.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ProjectManager.cpp b/src/ProjectManager.cpp
index cd4c2d2..e764f7b 100644
--- a/src/ProjectManager.cpp
+++ b/src/ProjectManager.cpp
@@ -874,7 +874,7 @@ bool ProjectManager::Save (bool as_template) {
 					if (!breakpoints[j]->enabled) CFG_BOOL_WRITE_DN("breakpoint_enabled",false);
 					if (breakpoints[j]->action) CFG_GENERIC_WRITE_DN("breakpoint_action",true);
 					if (breakpoints[j]->ignore_count) CFG_GENERIC_WRITE_DN("breakpoint_ignore",breakpoints[j]->ignore_count);
-					if (breakpoints[j]->annotation) CFG_GENERIC_WRITE_DN("breakpoint_annotation",mxUT::Text2Line(breakpoints[j]->annotation));
+					if (!breakpoints[j]->annotation.IsEmpty()) CFG_GENERIC_WRITE_DN("breakpoint_annotation",mxUT::Text2Line(breakpoints[j]->annotation));
 					if (breakpoints[j]->cond.Len()) CFG_GENERIC_WRITE_DN("breakpoint_condition",breakpoints[j]->cond);
 	//				breakpoint->enabled=true; // que hace esto aca?
 				}
-- 
2.16.2.windows.1

