From 218a681a8a3a2f9dbdc3f03610de226ff01642b3 Mon Sep 17 00:00:00 2001
From: Tim S <stahta01@users.sourceforge.net>
Date: Sat, 24 Mar 2018 07:59:45 -0400
Subject: [PATCH 2/2] msys2: ldd_cmd = "ldd"

---
 src/mxExeInfo.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/mxExeInfo.cpp b/src/mxExeInfo.cpp
index 8d2f7f3..4b01f3d 100644
--- a/src/mxExeInfo.cpp
+++ b/src/mxExeInfo.cpp
@@ -142,7 +142,7 @@ wxPanel *mxExeInfo::CreateGeneralPanel (wxNotebook *notebook) {
 
 wxPanel *mxExeInfo::CreateDependPanel (wxNotebook *notebook) {
 	CreatePanelAndSizer sizer(notebook);
-	sizer.BeginText( _if_win32("lsdeps","ldd") )
+	sizer.BeginText( "ldd" )
 		.Value( LANG(EXEINFO_WAIT_FOR_PARSER,"No se puede determinar esta informacion mientras el parser esta analizando fuentes") )
 		.ReadOnly().MultiLine().EndText(ldd_ctrl);
 	sizer.Set();
@@ -181,7 +181,7 @@ void mxExeInfo::OnTimer(wxTimerEvent &evt) {
 
 void mxExeInfo::UpdateTypeAndDeps ( ) {
 	if (ldd_ctrl) {
-		wxString ldd_cmd = _if_win32( config->GetZinjaiBinPath("lsdeps.exe"), "ldd" );
+		wxString ldd_cmd = "ldd.exe";
 		ldd_ctrl->SetValue(mxUT::GetOutput(ldd_cmd+" "+mxUT::Quotize(m_fname.GetFullPath()),true,false));
 	}
 	text_type->SetValue(mxUT::GetFileTypeDescription(m_fname.GetFullPath()));
-- 
2.16.2.windows.1

