From daf4825ccb90d479fbd6135fab16bdf3da73b328 Mon Sep 17 00:00:00 2001
From: Tim S <stahta01@users.sourceforge.net>
Date: Mon, 23 Jul 2018 21:15:47 -0400
Subject: [PATCH] Do not include boost m4 files

---
 configure.ac | 1 -
 1 file changed, 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 6d8df8288..15538af2c 100644
--- a/configure.ac
+++ b/configure.ac
@@ -14,7 +14,5 @@ AC_MSG_RESULT([${enable_flags_setting}])
 
 m4_include([m4/acinclude.m4])
 m4_include([m4/ax_cxx_compile_stdcxx.m4])
-m4_include([m4/ax_boost_base.m4])
-m4_include([m4/ax_boost_system.m4])
 
 CODEBLOCKS_CHECK_DEBUG
-- 
2.18.0.windows.1

